packages/integration-tests/tests/username-password-flow.test.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  createRequester,
+  fetchOidcConfig,
+  fetchTokenByAuthorizationCode,
+  generateSignInUri,
+  verifyAndParseCodeFromCallbackUri,
+} from '@logto/js';
+import got from 'got/dist/source';
+
+import api from '@/api';
+
+import { adminConsoleApplicationId, discoveryUrl, logtoUrl, redirectUri } from '../src/constants';
+import { createLogtoContext } from '../src/logto-context';
+import { extractCookie, generatePassword, generateUsername } from '../src/utils';
+
+describe('username and password flow', () => {
+  const {
+    account,
+    getInteractionCookie,
+    getState,
+    getCodeVerifier,
+    getCodeChallenge,
+    getAuthorizationCode,
+    getNextRedirectTo,
+    getAuthorizationEndpoint,
+    getTokenEndpoint,
+    initInteraction,
+    updateCookie,
+    setAuthorizationCode,
+    setNextRedirectTo,
+    setUpEndpoints,
+  } = createLogtoContext({
+    username: generatePassword(),
+    password: generateUsername(),
+  });
+
+  beforeAll(async () => {
+    await initInteraction();
```

Comment:
nit: how about
```
logtoContext.init();
```

or make it as a class: LogtoContext, and new an instance at the very beginning. 
and properties can be accessed directly. 

logtoContext.codeChallenge
logtoContext.state